server/index.js

```diff
@@ -106,6 +106,35 @@ app.get('/api/search/:keyword', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.patch('/api/events/:eventId', (req, res, next) => {
+  const eventId = parseInt(req.params.eventId, 10);
+  if (!eventId) {
+    throw new ClientError(400, 'eventId must be a positive integer');
+  }
+  const { title, description, timestamp, origin, destination, coords, notification, email } = req.body;
+  const sql = `
+       update "events"
+       set "email"        = $1,
+           "title"        = $2,
+           "description"  = $3,
+           "timestamp"    = $4,
+           "origin"       = $5,
+           "destination"  = $6,
+           "coords"       = $7,
+           "notification" = $8
+     where "userId" = $9 AND
+           "eventId" = $10
+    returning *;
+  `;
```

Comment:
If all of these values must be present, be sure to add validation. Otherwise, you can let them do a partial update to keep the previous value.

```sql
       update "events"
       set "email"        = coalesce($1, "email"),
           "title"        = coalesce($2, "title"),
           "description"  = coalesce($3, "description"),
           -- etc etc etc
     where "userId" = $9 AND
           "eventId" = $10
    returning *;
```